### PR TITLE
Implemented QuestionListEntry component

### PR DESCRIPTION
**Pull request by NMendini**
https://trello.com/c/kRZz7rVV

### CODE REVIEW

File: client/components/QuestionListEntry.jsx [hunk 1]
```diff
@@ -1,4 +1,5 @@
 import React from 'react';
+import Moment from 'react-moment';
 
 class QuestionListEntry extends React.Component {
```
Comment (Sam-Goldie) [inline client/components/QuestionListEntry.jsx:4]: Hmm...is there any reason this has to be a stateful component? I think it's best to make a it a function unless you need to store state on it. If you do that elsewhere, I'd change it there too.

File: client/components/QuestionList.jsx [hunk 2]
```diff
@@ -1,18 +1,12 @@
...
-        <div className="answer-row-name"><span>Name</span> — Post Date Goes Here <span className="helpful answer-click"> Helpful(number) </span><span className="not-helpful answer-click"> Not helpful(number) </span><span className="report-answer answer-click"> Report </span></div>
-        <div className="answer-row"><span className="answer-bold">A: </span><span>The point that pierces the pod is most likely clogged. There’s a tiny opening which needs to be cleared. I clean mine monthly by using a straightened paper clip. You can google how to clean your machine. Be sure to unplug your Keurig machine first.</span></div>
-        <div className="answer-row-name"><span>Name</span> — Post Date Goes Here <span className="helpful answer-click"> Helpful(number) </span><span className="not-helpful answer-click"> Not helpful(number) </span><span className="report-answer answer-click"> Report </span></div>
-      </div>
-      <button type="button" className="btn">Answer it</button>
-      <hr></hr>
+    <div>{
+      props.questions.map((item, i) =>
+        <QuestionListEntry question={item} key={i} answers={props.answers}/>
```
Comment (Sam-Goldie) [inline client/components/QuestionList.jsx:8]: I forgot to pass in the key to my own project! Thanks for reminding me!

File: client/components/QAndA.jsx [hunk 3]
```diff
@@ -0,0 +1,35 @@
+import React from 'react';
+import QuestionList from './QuestionList.jsx';
+
+class QAndA extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
```
Comment (Sam-Goldie) [inline client/components/QAndA.jsx:8]: Again, I'd convert this to a stateless function.

**Review by Sam-Goldie — COMMENTED** [review]
Not much to say. I'd check the statefulness of your components. Good work.

File: client/components/QuestionListEntry.jsx (hunk 1 above)
Comment (NMendini) [reply client/components/QuestionListEntry.jsx:4]: I appreciate the feedback! I was unsure of whether to do stateful or stateless as some of the functionality I'll need later on will probably use state. I think when I decide where I want to hold that state, I can put your advice to good use and convert the other components to stateless. Thank you!

**Pull request merged**